nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -453,6 +453,17 @@ sub in {
         }
     }
 
+    # Preserve umask (fmask, dmask) settings for vfat filesystems.
+    # (The default is to mount these world-readable, but that's a security risk
+    # for the EFI System Partition.)
+    if ($fsType eq "vfat") {
```

Comment:
This condition is somewhat strange. Maybe we need to make it more precise or more general?

Reply:
What do you have in mind? And, can we do that in a follow-up PR?

Reply:
In what way is the condition strange?

Reply:
vfat doesn't necessarily mean EFI partition. Also do partitions in other formats need this feature?

Reply:
Ah, I see. I thought about checking the mountpoint, but then again I didn't want to introduce weird/artificial restrictions in this tool either.

Having nixos-generate-config emit fmask= and dmask= on non-ESP mounts isn't necessarily a bad thing either. One could argue it's the correct thing to do in any case (preserving the permissions that was used at install time). The only thing I could think of was if we could detect when no umask was used for the mount and then _not_ emit fmask= and dmask= in hardware-configuration.nix. But I'm not sure where the default mount option should be read from, and so I don't think it's worth doing.

> Also do partitions in other formats need this feature?

I don't know. But this was specifically a fix for systemd-boot/EFI, and it only supports FAT32. (Also, we can always improve later.)